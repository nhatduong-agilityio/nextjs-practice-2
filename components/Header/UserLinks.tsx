import { Button, VStack } from '@chakra-ui/react'
import React from 'react'
import NextLink from 'next/link'

const UserLinks = () => (
  <VStack w='full' gap='20px'>
    <NextLink href='/' passHref>
      Shopping Cart
    </NextLink>
    <NextLink href='/' passHref>
      Wishlist
    </NextLink>
    <NextLink href='/' passHref>
      Sign Up
    </NextLink>
    <NextLink href='/' passHref style={{ width: '100%' }}>
      <Button
        borderRadius='10px'
        width='full'
        variant='secondary'
        height={['52px', '66px']}
        fontSize={['sm', 'md']}
      >
        Login
      </Button>
    </NextLink>
  </VStack>
)

export default UserLinks
