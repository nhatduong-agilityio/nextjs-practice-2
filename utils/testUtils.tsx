import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
