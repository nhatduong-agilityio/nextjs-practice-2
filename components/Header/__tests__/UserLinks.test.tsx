// Utils
import { render } from '@utils/testUtils'

// Components
import UserLinks from '../UserLinks'

describe('UserLinks', () => {
  it('Should UserLinks match Snapshot', () => {
    const { container } = render(<UserLinks />)

    expect(container).toMatchSnapshot()
  })
})
