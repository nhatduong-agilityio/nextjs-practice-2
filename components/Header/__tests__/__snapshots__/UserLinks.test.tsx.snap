// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLinks Should UserLinks match Snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-1trjysb"
  >
    <a
      href="/"
    >
      Shopping Cart
    </a>
    <a
      href="/"
    >
      Wishlist
    </a>
    <a
      href="/"
    >
      Sign Up
    </a>
    <a
      href="/"
      style="width: 100%;"
    >
      <button
        class="chakra-button css-1b61qg5"
        type="button"
      >
        Login
      </button>
    </a>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
